src/rules/valid-context-access.ts

```diff
@@ -0,0 +1,113 @@
+import { createRule } from "../utils"
+import { extractContextReferences } from "./reference-helpers/svelte-context"
+import type { TSESTree } from "@typescript-eslint/types"
+
+export default createRule("valid-context-access", {
+  meta: {
+    docs: {
+      description:
+        "context functions must be called during component initialization.",
+      category: "Possible Errors",
+      // TODO Switch to recommended in the major version.
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      unexpected:
+        "Do not call {{function}} except during component initialization.",
+    },
+    type: "problem",
+  },
+  create(context) {
+    const scopeManager = context.getSourceCode().scopeManager
+    const toplevelScope =
+      scopeManager.globalScope?.childScopes.find(
+        (scope) => scope.type === "module",
+      ) || scopeManager.globalScope
+
+    /** report ESLint error */
+    function report(node: TSESTree.CallExpression) {
+      context.report({
+        loc: node.loc,
+        messageId: "unexpected",
+        data: {
+          function:
+            node.callee.type === "Identifier"
+              ? node.callee.name
+              : "context function",
+        },
+      })
+    }
+
+    /** Get nodes where the variable is used */
+    function getReferences(id: TSESTree.Identifier | TSESTree.BindingName) {
+      const variable = toplevelScope?.variables.find((v) => {
+        if (id.type === "Identifier") {
+          return v.identifiers.includes(id)
+        }
+        return false
+      })
+      if (variable) {
+        return variable.references.filter((r) => r.identifier !== id)
+      }
+      return []
+    }
+
+    /** Let's lint! */
+    function doLint(
+      visitedCallExpressions: TSESTree.CallExpression[],
+      contextCallExpression: TSESTree.CallExpression,
+      currentNode: TSESTree.CallExpression,
+    ) {
+      let { parent } = currentNode
+      if (parent?.type !== "ExpressionStatement") {
```

Comment:
I suspect an if statement like the following will not work:

```js
if (hasContext("answer")) {
}
```
Could you add this test case?

Reply:
I added a basic test for if statement. I think it works fine.
Is this same understanding with you?

https://github.com/sveltejs/eslint-plugin-svelte/pull/480/commits/a86eac8312032e1008f84c67b2a9b59590d71bc6

Reply:
Sorry. I should have explained more.
I would like to add a test for if statements are used at the top level.

```svelte
<script>
  import { hasContext } from "svelte"

  if (hasContext("answer")) {
    console.log("The answer exist")
  }
</script>
```

Reply:
Oh OK! I will add a test.